DSWx/Reservoir/Time_Slider_Visualization.ipynb

```diff
@@ -0,0 +1,338 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Visualizing change in reservoir surface water extent using the OPERA DSWx product with a time slider\n",
+    "---\n",
+    "\n",
+    "**This notebook serves as a visualization tool using DSWx B01_WTR to illustrate change of Lake Mead, NV during 2022.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# notebook dependencies\n",
+    "import os\n",
+    "import math\n",
+    "import xarray as xr\n",
+    "import rasterio as rio\n",
+    "import rioxarray as rioxr\n",
+    "import hvplot.xarray\n",
+    "import geoviews as gv\n",
+    "import pyproj\n",
+    "from pyproj import Proj\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import geopandas as gpd\n",
+    "import holoviews as hv\n",
+    "import panel.widgets as pnw\n",
+    "import datetime\n",
+    "from datetime import datetime\n",
+    "\n",
+    "from bokeh.models import FixedTicker\n",
+    "hv.extension('bokeh')\n",
+    "\n",
+    "import warnings\n",
+    "warnings.filterwarnings('ignore')\n",
+    "\n",
+    "os.environ['AWS_NO_SIGN_REQUEST'] = 'YES'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "## **Data Information Input**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the code cell below, the user should specify the:\n",
+    "* Dates of interest <br>\n",
+    "* Data directory<br>\n",
+    "* Band of interest<br>\n",
+    "* Path to shapefile to create mask<br><br>\n",
+    "\n",
+    "**<font color='red'>Note: The cell below is the only code in the notebook that should be modified. </font>**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Access all 2022 provisional products from S3 bucket\n",
+    "data_dir = [match for match in sorted(open(\"aux_files/T11SQA_manifest.txt\",\"r\").read().split(\"\\n\")) if '2022' in match]\n",
```

Comment:
Very minor point, but there may/or may not be very fringe cases like this:
'//opera-pst-rs-pop1/products/OPERA_L3_DSWx-HLS_T11SQA_20210102T202201Z_20230118T204610Z_S2A_30_v0.0/OPERA_L3_DSWx-HLS_T11SQA_20210102T202201Z_20230118T204610Z_S2A_30_v0.0_'

Where this dataset corresponds to a date in `2021`, but would get filtered through as a `2022` date with this logic.

I'd instead advise something like this:
`data_dir = [match for match in sorted(open("aux_files/T11SQA_manifest.txt","r").read().split("\n")) if 'A_2022' in match]`

Where adding the `A_` in front prevents such a false match. I'd also advise on adding the `A` as there may be a scenario e.g. that you only want `2023` dates, but then everything on record as of present would be selected as it was all produced this year, and thus gets captured by the second part of the timestamp (see the example above)

Reply:
But I just though about another scenario where the string preceding the time may change. Thus, to future proof this, I suggest something along the lines of the following:
`data_dir2 = [match for match in sorted(open("aux_files/T11SQA_manifest.txt","r").read().split("\n")) if '2022' == match.split("_")[-5][:4]]`

In this case, I'm directly stripping out the time field we want, and looking at only the year. Though as a caveat, I should note that in the past I've admittedly annoyed people with list comprehension if it gets too complex, so you may consider translating this into a simple for loop. Your call though!

Reply:
I think your second approach is a better way to go about things, since I was originally trying to filter based on the date.  Constructing the list of file names already takes some finagling, so I think your suggestion doesn't make it that much more complicated